Remotion/Web/Development.WebTesting/HostingStrategies/DockerHostingStrategy.cs

```diff
@@ -57,49 +75,62 @@ public class DockerHostingStrategy : IHostingStrategy
           .Distinct(StringComparer.OrdinalIgnoreCase)
           .ToArray();
 
-      var configurationParameters = new IisDockerContainerConfigurationParameters(
-          absoluteWebApplicationPath,
-          port,
-          dockerImageName,
-          dockerIsolationMode,
-          hostname,
-          is32BitProcess,
-          mounts);
+      _dockerContainerWrapper = innerType switch
+      {
+        null => ThrowNoInnerTypeException(),
+        _ when innerType.Equals("iisExpress", StringComparison.OrdinalIgnoreCase) => CreateIisDockerContainer(),
+        _ when innerType.Equals("aspnetcore", StringComparison.OrdinalIgnoreCase) => CreateAspNetCoreDockerContainer(),
+        _ => ThrowNoInnerTypeException()
+      };
 
-      _iisDockerContainerWrapper = new IisDockerContainerWrapper(docker, configurationParameters);
-    }
+      DockerContainerWrapperBase CreateIisDockerContainer ()
+      {
+        var configurationParameters = new DockerContainerConfigurationParameters(
+            absoluteWebApplicationPath,
+            port,
+            dockerImageName,
+            dockerIsolationMode,
+            hostname,
+            is32BitProcess,
+            mounts);
 
-    /// <summary>
-    /// Constructor required for direct usage in <see cref="WebTestConfigurationSection"/>.
-    /// </summary>
-    /// <param name="testSiteLayoutConfiguration">The configuration of the used test site.</param>
-    /// <param name="properties">The configuration properties.</param>
-    [UsedImplicitly]
-    public DockerHostingStrategy ([NotNull] ITestSiteLayoutConfiguration testSiteLayoutConfiguration, [NotNull] IReadOnlyDictionary<string, string> properties)
-        : this(
-            testSiteLayoutConfiguration,
-            int.Parse(ArgumentUtility.CheckNotNull("properties", properties)["port"]!),
-            properties["dockerImageName"]!,
-            properties["dockerIsolationMode"],
-            TimeSpan.Parse(properties["dockerPullTimeout"]!),
-            properties["hostname"])
-    {
-      // TODO RM-8113: Guard used properties against null values.
+        return new IisDockerContainerWrapper(docker, configurationParameters);
+      }
+
+      DockerContainerWrapperBase CreateAspNetCoreDockerContainer ()
+      {
+        var configurationParameters = new AspNetDockerContainerConfigurationParameters(
+            absoluteWebApplicationPath,
+            port,
+            dockerImageName,
+            dockerIsolationMode,
+            hostname,
+            is32BitProcess,
+            mounts,
+            testSiteLayoutConfiguration.ProcessPath);
+
+        return new AspNetDockerContainerWrapper(docker, configurationParameters);
+      }
+
+      DockerContainerWrapperBase ThrowNoInnerTypeException ()
+      {
+        throw new InvalidOperationException($"Could not resolve inner Type '{innerType}'. (Possible values: 'iisExpress' or 'aspnetcore')");
```

Comment:
please check. I think this should be a ConfigurationException, but only if there's a) precendent here and b) ConfigurationException is actually available in .NET 8 without the extension packs.